addon/components/polaris-button.js

```diff
@@ -322,107 +318,122 @@ export default Component.extend({
     'monochrome',
     'size',
     'fullWidth',
-    'isIconOnly',
-    function() {
-      let {
-        class: externalClasses,
-        primary,
-        outline,
-        destructive,
-        isDisabled,
-        loading,
-        plain,
-        monochrome,
-        size,
-        fullWidth,
-        isIconOnly,
-      } = this.getProperties(
-        'class',
-        'primary',
-        'outline',
-        'destructive',
-        'isDisabled',
-        'loading',
-        'plain',
-        'monochrome',
-        'size',
-        'fullWidth',
-        'isIconOnly'
-      );
-      let classes = ['Polaris-Button'];
-
-      if (isPresent(externalClasses)) {
-        classes.push(externalClasses);
-      }
-
-      if (primary) {
-        classes.push('Polaris-Button--primary');
-      }
+    'isIconOnly'
+  ).readOnly())
+  get classes() {
+    let {
+      class: externalClasses,
+      primary,
+      outline,
+      destructive,
+      isDisabled,
+      loading,
+      plain,
+      monochrome,
+      size,
+      fullWidth,
+      isIconOnly,
+    } = this.getProperties(
```

Comment:
we could probably skip using `getProperties` and directly destructure from `this`

Reply:
ohh, indeed, we support `3.10+`...will update